identity-hub-did/src/main/java/org/eclipse/dataspaceconnector/identityhub/did/IdentityHubCredentialsVerifier.java

```diff
@@ -0,0 +1,107 @@
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.did;
+
+import com.nimbusds.jwt.SignedJWT;
+import org.eclipse.dataspaceconnector.iam.did.spi.credentials.CredentialsVerifier;
+import org.eclipse.dataspaceconnector.iam.did.spi.document.DidDocument;
+import org.eclipse.dataspaceconnector.identityhub.client.IdentityHubClient;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.result.AbstractResult;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Obtains and verifies credentials associated with a DID.
+ * The DID document contains an IdentityHub service, the IdentityHubCredentialsVerifier gets credentials from the
+ * IdentityHub instance and verifies the credentials.
+ */
+public class IdentityHubCredentialsVerifier implements CredentialsVerifier {
+
+    private static final String VERIFIABLE_CREDENTIALS_KEY = "vc";
+    private static final String CREDENTIALS_ID_KEY = "id";
+    private static final String IDENTITY_HUB_SERVICE_TYPE = "IdentityHub";
+    private final IdentityHubClient identityHubClient;
+    private final Monitor monitor;
+    private final SignatureVerifier signatureVerifier;
+
+    /**
+     * Create a new credential verifier that uses an Identity Hub
+     *
+     * @param identityHubClient IdentityHubClient.
+     */
+    public IdentityHubCredentialsVerifier(IdentityHubClient identityHubClient, Monitor monitor, SignatureVerifier signatureVerifier) {
+        this.identityHubClient = identityHubClient;
+        this.monitor = monitor;
+        this.signatureVerifier = signatureVerifier;
+    }
+
+    /**
+     * Get credentials from the IdentityHub of participant, verifies the credentials, and returns the successfully verified credentials.
+     * To be successfully verified, credentials needs to be signed by the specified issuer in the JWT.
+     *
+     * @param didDocument of a participant. The Did Document should contain an IdentityHub service.
+     * @return VerifiableCredentials.
+     */
+    @Override
+    public Result<Map<String, Object>> getVerifiedCredentials(DidDocument didDocument) {
+        var hubBaseUrl = getIdentityHubBaseUrl(didDocument);
+        if (hubBaseUrl.failed()) return Result.failure(hubBaseUrl.getFailureMessages());
+
+        var jwts = identityHubClient.getVerifiableCredentials(hubBaseUrl.getContent());
+        if (jwts.failed()) {
+            return Result.failure(jwts.getFailureMessages());
+        }
+        var verifiedClaims = jwts.getContent()
+                .stream()
+                .filter(signatureVerifier::isSignedByIssuer);
+        var claims = verifiedClaims.map(this::extractCredential)
```

Comment:
Shouldn't we also here filter for credentials that are about the caller subject? The identity hub might store VCs about other subjects